src/test/java/ca/corefacility/bioinformatics/irida/service/impl/unit/ProjectServiceImplTest.java

```diff
@@ -200,12 +203,15 @@ public void testAddSampleToProjectNoSamplePersistedInvalidSample() {
 
 		when(validator.validate(s)).thenReturn(violations);
 
-		projectService.addSampleToProject(p, s, true);
+		assertThrows(ConstraintViolationException.class, () -> {
+			projectService.addSampleToProject(p, s, true);
+		});
 
-		verifyNoInteractions(sampleRepository, psjRepository);
+		verifyNoInteractions(psjRepository);
+		verify(sampleRepository, never()).save(s);
```

Comment:
Prior implementation of the the test never hit `verifyNoInteractions` because the test would complete when `ConstraintViolationException` was thrown by `projectService.addSampleToProject(p, s, true)`.

`sampleRepository` was removed from `verifyNoInteractions` because it is interacted with on line 359 of `ProjectServiceImpl.java` specifically to search for an existing sample with that name in the project. I have changed this to ensure that the save method is never called with our sample, which is really what it should have been testing.

Reply:
Thanks for leaving this comment here and explaining it